### ISSUE TITLE
Improve TypeChecking with TypeScript

### ISSUE THREAD

**Issue by alexakasanjeev** [vonovak/react-navigation-header-buttons#107]
Hello, My project is in TypeScript and I'm trying to use correct types for all component

As in your **How to integrate in your project** step, I followed

```jsx
import * as React from 'react';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {
  HeaderButtons,
  HeaderButton,
  HeaderButtonProps,
  HeaderButtonsProps,
} from 'react-navigation-header-buttons';

// define IconComponent, color, sizes and OverflowIcon in one place
const CustomHeaderButton = (props: HeaderButtonProps) => (
  <HeaderButton
    IconComponent={MaterialIcons}
    iconSize={23}
    color="black"
    {...props}
  />
);

export const CustomHeaderButtons = (props: HeaderButtonsProps) => {
  return (
    <HeaderButtons HeaderButtonComponent={CustomHeaderButton} {...props} />
  );
};
export { Item as CustomHeaderItem } from 'react-navigation-header-buttons';
```

And this is the `type` definition of `HeaderButtonsProps` in your code

```ts
export interface HeaderButtonsProps {
  /**
   * Whether the `HeaderButtons` are on the left from header title.
   * @default false
   */
  left?: boolean;
  /**
   * Component that renders the buttons.
   *
   * Typically, you'll want to provide a component that wraps `HeaderButton`
   * provided by this package, as seen in the quick example.
   * However, you're free to use your own component (see `HeaderButton` for reference).
   */
  HeaderButtonComponent?: ComponentType<any>;
}
```

But in code I'm getting this error when i use like this

```jsx
<CustomHeaderButtons>
    <CustomHeaderItem
              title={translate('common.menu')}
              iconName="menu"
              onPress={navigation.toggleDrawer}
      />
</CustomHeaderButtons>
```

![Screenshot from 2020-12-26 12-51-28](https://user-images.githubusercontent.com/13250741/103147213-37efa800-4779-11eb-82bb-b90564234a14.png)

**Comment by vonovak** [comment]
Hello, thanks for reporting. Can you please open a PR with this fix? Thank you.

**Comment by alexakasanjeev** [comment]
Hi @vonovak sure, I will, but am I right(what I propose), I'm not 100% sure ?

**Comment by vonovak** [comment]
I tend to believe you are :) try if it fixes the problem and if it does, I'll merge. Thanks!

**Comment by alexakasanjeev** [comment]
I added the key inside the local copy of the library(by going to node_modules > react-navigation-header-button > index.d.ts), and the error went away :)

**Comment by vonovak** [comment]
closed in #108
